pkg/controllers/awsloadbalancercontroller/controller.go

```diff
@@ -79,49 +80,65 @@ type AWSLoadBalancerControllerReconciler struct {
 func (r *AWSLoadBalancerControllerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	_ = log.FromContext(ctx)
 
-	lbController := &albo.AWSLoadBalancerController{}
-	if err := r.Client.Get(ctx, req.NamespacedName, lbController); err != nil {
-		if errors.IsNotFound(err) {
-			return ctrl.Result{}, nil
-		}
-		return ctrl.Result{}, fmt.Errorf("failed to get AWSLoadBalancerController %s: %w", req, err)
+	lbController, exists, err := r.getAWSLoadBalancerController(ctx, req.Name)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get AWSLoadBalancerController %s: %w", req.Name, err)
+	}
+	if !exists {
+		return ctrl.Result{}, nil
 	}
 
 	servingSecretName := fmt.Sprintf("%s-serving-%s", controllerResourcePrefix, lbController.Name)
 
 	// if the processed subnets have not yet been written into the status or if the tagging policy has changed then update the subnets
 	if lbController.Status.Subnets == nil || (lbController.Spec.SubnetTagging != lbController.Status.Subnets.SubnetTagging) {
-		err := r.tagSubnets(ctx, lbController)
+		internalSubnets, publicSubnets, taggedSubnets, untaggedSubnets, err := r.tagSubnets(ctx, lbController)
```

Comment:
Could you explain what you mean in more detail. Perhaps with an example.

Reply:
We cannot set a watch on the subnets like Kubernetes resources and then reconcile them. There is polling but that is not an option because of the tight rate-limits for API calls. If the user manually changes stuff that's on them.